### PR TITLE
Update checker

### PR DESCRIPTION
**Pull request by josh-richardson**
Draft PR for the update checker. Should write some tests before this all gets merged.

### CODE REVIEW

File: src/main/java/org/web3j/console/Runner.java [hunk 1]
```diff
@@ -43,6 +45,18 @@
     public static void main(String[] args) throws Exception {
         System.out.println(LOGO);
 
+        CliConfig config = CliConfig.getConfig();
+        Updater updater = new Updater(config);
+        if (!updater.updateCurrentlyAvailable()) {
+            new Thread(updater::onlineUpdateCheck).start();
```
Comment (cfelde) [inline src/main/java/org/web3j/console/Runner.java]: Maybe we should run this even if updater.updateCurrentlyAvailable() returns true?

Comment (josh-richardson) [reply src/main/java/org/web3j/console/Runner.java]: Fixed this

**Comment by josh-richardson** [comment]
Latest commit refactors significantly and adds a (rather large) test.

**Comment by mordechaim** [comment]
Please check out [update4j](https://github.com/update4j/update4j)

File: src/main/resources/Template.java [hunk 2]
```diff
@@ -1,8 +1,7 @@
 package <package_name>;
 public class <project_name> {
 
+    public static void main(String[] args) {
 
-public static void main(String[]args) {
-
-        }
+    }
 }
```
Comment (josh-richardson) [inline src/main/resources/Template.java]: Not strictly relevant to this PR, but fixes the issue where the generator generates Java with weird indentation.

**Comment by josh-richardson** [comment]
This is ready for review, but should not be merged until I've got the version checking service up & running on k8s.

**Comment by josh-richardson** [comment]
This should now be ready to merge after review.

**Comment by cfelde** [comment]
Besides failing tests, LGTM

**Pull request merged**